### ISSUE TITLE
introduce brick-layering as from CNC Kitchen Video

### ISSUE THREAD

**Issue by berberic2** [Ultimaker/Cura#18353]
### Is your feature request related to a problem?

Should improve layer adhesion

### Describe the solution you'd like

Please take a look at the excellent video from CNC Kitchen, where he tests what he calls brick layerning.
Effectifly moving every other perimeter-line up ½layer-height.

https://www.youtube.com/watch?v=5hGm6cubFVs

### Describe alternatives you've considered

none

### Affected users and/or printers

Would be an option.

**Comment by discite** [comment]
This will introduce a 10% strength, definetly interested in seeing it on the future

**Comment by lukamilosevic** [comment]
+1

**Comment by berberic2** [comment]
Take a look at https://www.youtube.com/watch?v=9IdNA_hWiyE  to shed some light at the patent situation.

**Comment by MaximeGranizo** [comment]
This would be a very useful feature! Improving layer adhesion is crucial for print quality, and incorporating the brick layering method as demonstrated in the CNC Kitchen video seems like a practical and innovative solution. It would be great to see this as an option for users looking to enhance their prints!

**Comment by anubis71x** [comment]
+1

**Comment by tilmaen** [comment]
I would totally love this feature.
The european Patent hasn't been filed yet. it seems it is stuck because of a more thorough examination process. 3rd-Party observations have been filed (and acknowledged), also the examiner seemed to have found several issues of "non novelty", Also the bungled up references to the prior Patents were critizized by the EPC officer (as opposed to the USTPO).
The correspondence is pretty interesting [https://register.epo.org/application?number=EP21175830&lng=en&tab=doclist](url)
i have a friend who works there (different department) i might ask him, whether he can tell me something about that patent application.

**Comment by mahtDFR** [comment]
Looks like it's been sort of implemented now...
https://www.youtube.com/watch?v=EqRdQOoK5hc

**Comment by BarsMonster** [comment]
> The european Patent hasn't been filed yet. it seems it is stuck because of a more thorough examination process. 3rd-Party observations have been filed (and acknowledged), also the examiner seemed to have found several issues of "non novelty", Also the bungled up references to the prior Patents were critizized by the EPC officer (as opposed to the USTPO). The correspondence is pretty interesting [https://register.epo.org/application?number=EP21175830&lng=en&tab=doclist](url) i have a friend who works there (different department) i might ask him, whether he can tell me something about that patent application.

Apparently, there is exact match with expired Stratasys patent, making all new patents on brick layers invalid. 
https://patents.google.com/patent/US5653925A/en (expired in 2015)
https://www.youtube.com/watch?v=9IdNA_hWiyE

**Comment by BarsMonster** [comment]
@Flopalop2 Yes, regardless of patent validity (i.e. is there sufficient prior art) to protect Ultimaker it is important NOT to introduce "Simplified brick layers" feature as a checkbox. Having separate outer wall configuration thickness for odd and even layers, where end user makes a decision how they want to print - is a safer way to go.

Also, my proposed solution does not look exactly like Fig5. if outer surface is on the left, it is proposed to allow difference on thickness on the inner side, which enhances adhesion to infill (which is not covered in the patent).

Picture from @GregValiant  :

![Image](https://github.com/user-attachments/assets/a74074da-f627-4ef8-a0e3-009f880ccec3)

Further changes can be made to increase "legal separation" from the patent: have outermost perimeter thickness identical, and only introduce thickness variability on second perimeter and deeper inside the print (to increase strength without affecting fine surface details).

**Comment by Flopalop2** [comment]
That could totally work. Adding configuration to any implementation would allow for looser interpretations of the patentable idea. The alternate extra wall and connect infill would be replaced with the offset layers. Good idea. I'm hoping Ultimaker steps up and allows one of these options (or both). Until then, I'm just trying to keep these threads alive haha.

**Comment by BarsMonster** [comment]
For these interested in intesting, in #20189 there is already experimental build working.

**Comment by bondarchook** [comment]
New insights into the "brick layers" technique from CNC Kitchen:
https://www.youtube.com/watch?v=dDgA51zdfLc

TLDR: need to increase flow while using the "brick layers" technique for maximum strength.

**Comment by Flopalop2** [comment]
I watched the video yesterday, and I can't help but think the increased flow is only super applicable if using post processing. If we can get brick layers built in, decreasing distance between walls would be a better solution than increasing flow. AFAIK this is not a setting in Cura, otherwise this would be as simple as increasing wall overlap and using the post processing script for brick layers. This would decrease total wall thickness, but increase strength while retaining dimensional accuracy. Changing flow messes with a lot of things like pressure advance/ linear advance that I'd rather not have to worry about, even if only the inner walls' flow was changed.

**Comment by Flopalop2** [comment]
Now, having extra flow is still important for closing both horizontal and vertical gaps if you want a really solid part, since perfect flow the nozzle puts out a rounded bead, and round objects don't mesh perfectly so you have to increase the flow and add part size changes to compensate.

**Comment by cfbrolley** [comment]
> If we can get brick layers built in, decreasing distance between walls would be a better solution than increasing flow. AFAIK this is not a setting in Cura

 decrease line width but then then increase flow so that the volumetric flow remains the same as it was. e.g instead of 0.4mm line width, setting a 0.35mm line width with 114.3% flow means that the slicer places the lines 0.35mm apart, but it's flowing the same volume as if it was printing a 0.4mm line, so it effectively prints a 0.4mm line that overlaps slightly. Can do it with more or less overlap too - 0.3mm line width + 133.4% flow, 0.25 line width +160% flow, etc...

since you can specify line width and flow seperately for infill lines, top/bottom lines, support lines, etc - you would be able to set those to 0.4mm + 100% flow, so things like PA or LA should remain the same, becuase the volume out the nozzle remains exactly the same. basically just "tricking" the slicer in to printing 0.4mm walls, but placed closer together.
